tools/lsp/preview/ui.rs

```diff
@@ -234,147 +239,114 @@ fn map_property_declaration(
+                        .and_then(|v| i32::try_from(v).ok())
+                        .unwrap_or_default();
                 }
             }
-        };
-        is_first = false;
-    }
-
-    last_was_quote.then_some(result)
-}
-
-fn simplify_value(
-    document_cache: &common::DocumentCache,
-    property_type: &str,
-    code_block_or_expression: &Option<properties::CodeBlockOrExpression>,
-) -> SimpleValueData {
-    let property_value = if let Some(cboe) = code_block_or_expression {
-        cboe.text().to_string()
-    } else {
-        String::new()
-    };
-
-    if property_type == "bool"
-        && (property_value == "true" || property_value == "false" || property_value.is_empty())
-    {
-        let value: SharedString =
-            if property_value == "true" { "true".into() } else { "false".into() };
-        return SimpleValueData {
-            widget: "bool".into(),
-            meta_data: Rc::new(VecModel::from(vec![value])).into(),
-            visual_items: Rc::new(VecModel::default()).into(),
-        };
-    } else if property_type == "string" {
-        if let Some(simple) = simplify_string(&property_value) {
-            return SimpleValueData {
-                widget: "string".into(),
-                meta_data: Rc::new(VecModel::from(vec![simple.into()])).into(),
-                visual_items: Rc::new(VecModel::default()).into(),
-            };
-        }
-    } else if let Some(property_type) = property_type.strip_prefix("enum ") {
-        if let i_slint_compiler::langtype::Type::Enumeration(enumeration) =
-            &document_cache.global_type_registry().lookup(property_type)
-        {
-            let short_property_value = property_value
-                .strip_prefix(&format!("{property_type}."))
-                .unwrap_or(&property_value);
-            let type_name: SharedString = property_type.into();
-            let default_value: SharedString = enumeration.default_value.to_string().into();
-            let current_value = enumeration
-                .values
-                .iter()
-                .position(|v| v == short_property_value)
-                .map(|p| SharedString::from(p.to_string()))
-                .unwrap_or_else(|| default_value.clone());
-            let visual_values: Vec<_> = enumeration.values.iter().map(SharedString::from).collect();
-            return SimpleValueData {
-                widget: "enum".into(),
-                meta_data: Rc::new(VecModel::from(vec![type_name, default_value, current_value]))
-                    .into(),
-                visual_items: Rc::new(VecModel::from(visual_values)).into(),
-            };
         }
+        _ => {}
     }
 
-    SimpleValueData {
-        widget: SharedString::new(),
-        meta_data: Rc::new(VecModel::default()).into(),
-        visual_items: Rc::new(VecModel::default()).into(),
+    PropertyValue {
+        kind,
+        value_bool,
+        value_string: value_string.into(),
```

Comment:
(you could use SharedString directly so you don't need to allocate a String.)

Reply:
I was considering that, but found it a bit cumbersome as I update that value in several places and SharedString is less nice to use.

I'll leave that out for now.